kinto/core/resource/__init__.py

```diff
@@ -148,6 +161,14 @@ class UserResource:
     """Schema to validate records."""
 
     def __init__(self, request, context=None):
+
+        self.request = request
+        self.context = context
+
+        # XXX: Check if the path is valid before doing anything. This have to be done
+        # before fetching any model, so we can't wait for regular colander validation.
+        self._raise_400_if_invalid_path()
```

Comment:
If that helps we could make `self.model` lazy, ie. instantiate at first use. Which should be late enough to let colander path validation happen before?